tests/test_gaussian.py

```diff
@@ -15,265 +15,502 @@
+        assert np.allclose(res, sf_res, atol=tol, rtol=0)
+
+    def test_interferometer(self, tol):
+        """Test that the Interferometer gate works correctly"""
+        U = np.array(
+            [
+                [0.83645892 - 0.40533293j, -0.20215326 + 0.30850569j],
+                [-0.23889780 - 0.28101519j, -0.88031770 - 0.29832709j],
+            ]
+        )
+
+        wires = [0, 1]
+
+        gate_name = "Interferometer"
+        operation = qml.ops.Interferometer
+
+        dev = qml.device("strawberryfields.gaussian", wires=2)
+
+        sf_operation = dev._operation_map[gate_name]
+
+        assert dev.supports_operation(gate_name)
+
+        @qml.qnode(dev)
+        def circuit(*args):
+            qml.TwoModeSqueezing(0.1, 0, wires=[0, 1])
+            operation(*args, wires=wires)
+            return qml.expval(qml.NumberOperator(0)), qml.expval(qml.NumberOperator(1))
+
+        res = circuit(U)
+        sf_res = SF_gate_reference(sf_operation, wires, U)
+        assert np.allclose(res, sf_res, atol=tol, rtol=0)
+
+    def test_displaced_squeezed_state(self, tol):
+        """Test that the DisplacedSqueezedState gate works correctly"""
+        a = 0.312
+        b = 0.123
+        c = 0.532
+        d = 0.124
+
+        wires = [0]
+
+        gate_name = "DisplacedSqueezedState"
+        operation = qml.DisplacedSqueezedState
+
+        dev = qml.device("strawberryfields.gaussian", wires=2)
+
+        sf_operation = dev._operation_map[gate_name]
+
+        assert dev.supports_operation(gate_name)
+
+        @qml.qnode(dev)
+        def circuit(*args):
+            qml.TwoModeSqueezing(0.1, 0, wires=[0, 1])
+            operation(*args, wires=wires)
+            return qml.expval(qml.NumberOperator(0)), qml.expval(qml.NumberOperator(1))
+
+        res = circuit(a, b, c, d)
+        sf_res = SF_gate_reference(sf_operation, wires, a * np.exp(1j * b), c, d)
+        assert np.allclose(res, sf_res, atol=tol, rtol=0)
+
+
+class TestUnsupported:
+    """Test that an error is raised for gates and observables that are
+    unsupported."""
+
+    @pytest.mark.parametrize(
+        "gate_name,pennylane_gate", unsupported_one_mode_single_real_parameter_gates
+    )
+    def test_unsupported_one_mode_single_real_parameter_gates(self, gate_name, pennylane_gate, tol):
+        """Test those unsupported gates for the gaussian simulator that take a
+        single real parameter and act on one mode """
+        dev = qml.device("strawberryfields.gaussian", wires=2)
+        a = 0.312
+
+        operation = pennylane_gate
+        wires = [0]
+
+        @qml.qnode(dev)
+        def circuit():
+            operation(a, wires=wires)
             return qml.expval(qml.NumberOperator(0))
 
-        self.assertAlmostEqual(circuit(1), 1, delta=self.tol)
+        with pytest.raises(
+            qml.DeviceError,
+            match="Gate {} not supported " "on device strawberryfields.gaussian".format(gate_name),
+        ):
+            circuit()
```

Comment:
Nitpick: since these tests are testing an error that is being raised in PennyLane core, maybe it's not strictly necessary to test them here? But fine to leave them in for completeness!

Reply:
That's true! However, they are raising it on a custom device that was defined in this plugin, so perhaps they belong here?